android-o/AndroidPlayLocation/LocationUpdatesForegroundService/LocationUpdatesForegroundService/LocationUpdatesService.cs

```diff
@@ -0,0 +1,349 @@
+	 */
+	[Service(Label = "LocationUpdatesService", Enabled = true, Exported = true)]
+	[IntentFilter(new string[] { "com.xamarin.LocationUpdatesForegroundService.LocationUpdatesService" })]
+	public class LocationUpdatesService : Service
+	{
+		const string LocationPackageName = "com.google.android.gms.location.sample.locationupdatesforegroundservice";
+
+		const string Tag = "LocationUpdatesService";
+
+		public const string ActionBroadcast = LocationPackageName + ".broadcast";
+
+		public const string ExtraLocation = LocationPackageName + ".location";
+		const string ExtraStartedFromNotification = LocationPackageName + ".started_from_notification";
+
+		IBinder Binder;
+
+		/**
+	     * The desired interval for location updates. Inexact. Updates may be more or less frequent.
+	     */
+		const long UpdateIntervalInMilliseconds = 10000;
+
+		/**
+		 * The fastest rate for active location updates. Updates will never be more frequent
+		 * than this value.
+		 */
+		const long FastestUpdateIntervalInMilliseconds = UpdateIntervalInMilliseconds / 2;
+
+		/**
+		 * The identifier for the notification displayed for the foreground service.
+		 */
+		const int NotificationId = 12345678;
+
+		/**
+		 * Used to check whether the bound activity has really gone away and not unbound as part of an
+		 * orientation change. We create a foreground service notification only if the former takes
+		 * place.
+		 */
+		bool ChangingConfiguration = false;
+
+		NotificationManager NotificationManager;
+
+		/**
+		 * Contains parameters used by {@link com.google.android.gms.location.FusedLocationProviderApi}.
+		 */
+		LocationRequest LocationRequest;
+
+		/**
+		 * Provides access to the Fused Location Provider API.
+		 */
+		FusedLocationProviderClient FusedLocationClient;
+
+		/**
+		 * Callback for changes in location.
+		 */
+		LocationCallback LocationCallback;
+
+		Handler ServiceHandler;
+
+		/**
+		 * The current location.
+		 */
+		Location Location;
+
+		public LocationUpdatesService()
+		{
+			Binder = new LocationUpdatesServiceBinder(this);
+		}
+
+		class LocationCallbackImpl : LocationCallback
+		{
+			public LocationUpdatesService Service { get; set; }
+			public override void OnLocationResult(LocationResult result)
+			{
+				base.OnLocationResult(result);
+				Service.OnNewLocation(result.LastLocation);
+			}
+		}
+
+		public override void OnCreate()
+		{
+			FusedLocationClient = LocationServices.GetFusedLocationProviderClient(this);
+
+			LocationCallback = new LocationCallbackImpl { Service = this };
+
+			CreateLocationRequest();
+			GetLastLocationAsync();
+
+			HandlerThread handlerThread = new HandlerThread(Tag);
+			handlerThread.Start();
+			ServiceHandler = new Handler(handlerThread.Looper);
+			NotificationManager = (NotificationManager) GetSystemService(NotificationService);
+		}
+
+		public override StartCommandResult OnStartCommand(Android.Content.Intent intent, StartCommandFlags flags, int startId)
```

Comment:
No need to fully quality `Intent` here, as `Android.Content.` is previously declared in a using statement.

Reply:
Right, VS autocompleted it and didn't note

Reply:
**Done**